src/main/java/ay2122s1_cs2103t_w16_2/btbb/storage/JsonAdaptedOrder.java

```diff
@@ -72,6 +105,60 @@ public Order toModelType() throws IllegalValueException {
         }
         final Address modelClientAddress = new Address(clientAddress);
 
-        return new Order(modelClientName, modelClientPhone, modelClientAddress);
+        if (recipeName == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, GenericString.getMessageConstraints("Recipe Name")
+            ));
+        }
+        if (!GenericString.isValidGenericString(recipeName)) {
+            throw new IllegalValueException(GenericString.getMessageConstraints("Recipe Name"));
+        }
+        final GenericString modelRecipeName = new GenericString(recipeName);
+
+        final List<Ingredient> ingredients = new ArrayList<>();
+        for (JsonAdaptedIngredient ingredient : recipeIngredients) {
+            Ingredient i = ingredient.toModelType();
+
+            boolean hasValidName = GenericString.isValidGenericString(i.getName().toString());
+            boolean hasValidQuantity = Quantity.isValidQuantity(i.getQuantity().toString());
+            boolean hasValidUnit = GenericString.isValidGenericString(i.getUnit().toString());
```

Comment:
This isn't necessary right? `ingredient.toModelType()` already does the check. The only difference would be isValidQuantity-used for recipeIngredients vs isValidInternalQuantity-for Inventory